File: src/main/java/fr/khalypso/statesearchlib/model/Heuristic.java
package fr.khalypso.statesearchlib.model;

/**
 * Heuristic class model.
 */
public interface Heuristic {


	/**
	 * Estimate a state value.
	 *
	 * @param state
	 * 	the state to estimate
	 * @return the estimation in an integer value
	 */
	int eval(State state);
}

File: library/src/main/kotlin/fr/khalypso/statesearchlib/model/Heuristic.kt
package fr.khalypso.statesearchlib.model

/**
 * Heuristic class model.
 */
interface Heuristic {
    /**
     * Estimate a state value.
     *
     * @param state
     * the state to estimate
     * @return the estimation in an integer value
     */
    fun eval(state: State): Int
}
